vault/request_handling.go

```diff
@@ -589,8 +589,12 @@ func (c *Core) handleCancelableRequest(ctx context.Context, req *logical.Request
 				token, err = c.CheckSSCToken(ctx, token.(string), c.isLoginRequest(ctx, req), c.perfStandby)
 
 				// If we receive an error from CheckSSCToken, we can assume the token is bad somehow, and the client
-				// should receive a 403 bad token error like they do for all other invalid tokens.
+				// should receive a 403 bad token error like they do for all other invalid tokens, unless the error
+				// specifies that we should forward the request or retry the request.
 				if err != nil {
+					if strings.Contains(err.Error(), logical.ErrPerfStandbyPleaseForward.Error()) || strings.Contains(err.Error(), logical.ErrMissingRequiredState.Error()) {
```

Comment:
I was thinking the code could parallel how we throw status codes. Though I don't think we're doing it in the `CheckSSCT` function, we oftentimes add a prefix to the error so as to add more context -- the prefix also helps signify where in the request path the error was being thrown from. The status code check looks to see if the logical error is contained in the final error message. I wanted the code to encourage such prefixes, and also to avoid running the potential risk of losing the error due to such prefixes, which is why I went with the `.Contains` route.

Reply:
Summarizing our conversation -- the errors that are thrown use the `%w` directive to wrap errors, so `.Is` will work. We aren't in the right place in the code path to worry about altered strings in `logical.ErrorResponse`, but even if we were at that level we want to avoid inspecting the string contents of `logical.ErrorResponse` and focus on the contents of the thrown error instead. 

Will update the code to use `errors.Is`.